portal-impl/src/com/liferay/portlet/messageboards/service/impl/MBMessageLocalServiceImpl.java

```diff
@@ -1979,11 +1979,13 @@ protected String getDiscussionMessageSubject(String subject, String body)
 
 		subject = HtmlUtil.extractText(body);
 
-		if (subject.length() <= 50) {
-			return subject;
+		int pos = Math.min(subject.length(), 50);
+
+		if (subject.length() > pos) {
+			return subject.substring(0, pos) + StringPool.TRIPLE_PERIOD;
```

Comment:
According to the LPS ticket and the CS engineer the first 50 characters need to be shown. That is why I changed the substring method.